.github/workflows/ci.yml

```diff
@@ -62,43 +62,69 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        container-runtime: [focal, buster-slim]
-        runtime: [linux-x64]
+        container-runtime: [focal, buster-slim, alpine]
     runs-on: ubuntu-latest
     name: ci/github/build-${{ matrix.container-runtime }}
     steps:
     - name: Checkout
       uses: actions/checkout@v2
-    - shell: bash
-      run: |
-        git fetch --prune --unshallow
+      with:
+        fetch-depth: 0
     - name: Install net5.0
       uses: actions/setup-dotnet@v1
       with:
         dotnet-version: 5.0.x
+    - name: Setup QEMU
+      uses: docker/setup-qemu-action@v1
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v1
+    - name: Login to GitHub Container Registry
+      uses: docker/login-action@v1
+      with:
+        registry: ghcr.io
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
     - name: Calculate Version
       run: |
         dotnet tool restore
         version=$(dotnet tool run minver -- --tag-prefix=oss-v)-${{ matrix.container-runtime }}
         echo "VERSION=${version}" >> $GITHUB_ENV
+    - name: Setup Variables
+      id: variables
+      uses: kanga333/variable-mapper@master
+      with:
+        key: "${{ matrix.container-runtime }}"
+        map: |
+          {
+            "focal": {
+              "runtime": "linux-x64"
+            },
+            "buster-slim": {
+              "runtime": "linux-x64"
+            },
+            "alpine": {
+              "runtime": "alpine-x64"
+            }
+          }
+        export_to: output
     - name: Build
-      run: |
-        docker build \
-          --tag eventstore \
-          --build-arg RUNTIME=${{ matrix.runtime }} \
-          --build-arg CONTAINER_RUNTIME=${{ matrix.container-runtime }} \
-          .
+      uses: docker/build-push-action@v2
+      with:
+        context: .
+        load: true
+        tags: eventstore
```

Comment:
@thefringeninja I may be missing something - does `build-args` need to be specified here (and below in `Build Test Container` action) to pass in `matrix.runtime` and `matrix.container-runtime` or is it passed automatically?

Reply:
it does, I completely missed that.